torch/functional.py

```diff
@@ -170,14 +189,20 @@ def lu_unpack(LU_data, LU_pivots, unpack_data=True, unpack_pivots=True):
             P = torch.eye(m, device=LU_data.device, dtype=LU_data.dtype) \
                      .expand(shape[:-1] + (m,)) \
                      .clone(memory_format=torch.contiguous_format)
-            for idx in product(*map(lambda x: list(range(x)), shape[:-2])):
-                final_order = list(range(m))
-                for k, j in enumerate(LU_pivots_zero_idx[idx]):
+
+            # TODO: rewrite when TorchScript supports product and map as
+            # product(*map(lambda x: list(range(x)), shape[:-2]))
+            indices = indices_product(shape[:-2])
+            for idx in indices:
+                final_order = [i for i in range(m)]  # noqa: C416 TODO: rewrite as list(range(m))
+                for k, j in enumerate(index_tensor_with_indices_list(LU_pivots_zero_idx, idx)):
                     final_order[k], final_order[j] = final_order[j], final_order[k]
-                P[idx] = P[idx].index_select(1, torch.as_tensor(final_order, device=LU_pivots.device))
+                # TODO: remove index_tensor_with_indices_list when TorchScript supports indexing Tensor with list
```

Comment:
Can you file an issue for this and link it in the comment?

Reply:
when we had similar issues we just closed them - https://github.com/pytorch/pytorch/issues/32246, https://github.com/pytorch/pytorch/issues/27493